FindWoboq.cmake.in

```diff
@@ -0,0 +1,126 @@
+        PATHS @CMAKE_INSTALL_PREFIX@/bin
+        NO_DEFAULT_PATH
+        "create index")
+    # the static data-files:
+    SET(WOBOQ_DATA  "@CMAKE_INSTALL_PREFIX@/share/woboq/data" CACHE INTERNAL "woboq data")
+    # and this is where we ouput into. this is a fixed location.
+    SET(WOBOQ_OUTDIR "${CMAKE_BINARY_DIR}/woboq" CACHE INTERNAL "woboq outdir")
+    # because of using the cached variable, this message will only appear once,
+    # even when the "find_package()" macro is called multiple times
+    message("-- woboq found in '@CMAKE_INSTALL_PREFIX@', outputting into '${WOBOQ_OUTDIR}'")
+endif()
+
+# the global target to tie everything together by parsing all output into the
+# various index.html -- seems to mostly work.
+#
+# splitting the "data+index" step from the actual "generator" allows to add
+# content from multiple targets (invocations of the generator) into one
+# browsable "database" with one "data" folder in the root of the dir-tree
+if(NOT TARGET woboq)
+    add_custom_target(woboq
+        COMMAND cp -vr ${WOBOQ_DATA} ${WOBOQ_OUTDIR}
+        COMMAND ${WOBOQ_INDEXER} ${WOBOQ_OUTDIR} -d /data
+        COMMENT "provide static data and configure index.html"
+        )
+endif()
+
+# also cleanup the woboq-folder when asked to:
+set_property(DIRECTORY APPEND PROPERTY ADDITIONAL_MAKE_CLEAN_FILES
+    ${WOBOQ_OUTDIR}
+    )
+
+# a nice convenience target, which provides a local webserver to test the
+# result.  serves directly from the output-folder:
+if(NOT TARGET woboq-serve)
+    add_custom_target(woboq-serve
+        COMMAND python -m SimpleHTTPServer
+        WORKING_DIRECTORY ${WOBOQ_OUTDIR}
+        COMMENT "serving woboq from '${WOBOQ_OUTDIR}' at http://localhost:8000"
+        DEPENDS woboq
+        )
+endif()
+
+# this internal function executes woboq for _one_ target, into a subdir of
+# WOBOQ_OUTDIR:
+function(_woboqCreateNewGeneratorTarget)
+    set(options )
+    set(multiValueArgs )
+    set(oneValueArgs TARGET)
+    cmake_parse_arguments(_woboq "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN} )
+
+    # create function-local variables:
+    set(trgt ${_woboq_TARGET})
+    # the outdir cannot be overridden from the outside... in theory yes, but
+    # does not play well with the global "woboq" and "woboq-serve" targets
+    set(outdir ${WOBOQ_OUTDIR})
+
+    # this will (?) retrieve where the CMakeLists of this target is located. I
+    # think this is what the second part of the "-p" switch needs...
+    get_property(srcdir TARGET ${trgt} PROPERTY SOURCE_DIR)
+    # this is the "woboq" project name. we could add a version string...
+    #
+    # TODO: add current git-sha1 as revision
+    set(srcprjct ${trgt}:${srcdir})
+
+    # add a second "project" for files generated inside
+    # the global build directory. this will pout all generated file into a
+    # parallel page-tree in the served html.
+    get_property(blddir TARGET ${trgt} PROPERTY BINARY_DIR)
+    set(bldprjct ${trgt}/build:${blddir})
```

Comment:
But i don't think "build"  should be hard-coded here.

Note that because it is a sub directory, it will simply appear in a subdirectory of the project as normal .